### ISSUE TITLE
Maintainership change?

### ISSUE THREAD

**Issue by ted-gould** [xteve-project/xTeVe#320]
Hello @xteve-project it seems like you've lost interest in xTeVe, no judgement. I'm currently using it for some IPTV streams and I would like to see some of the various patches and improvements merged in. Would you consider a new maintainer? I'd be willing to step in and do that.

Thanks, Ted.

**Comment by deku-m** [comment]
Maybe you can ask take https://github.com/SCP002/xTeVe updates in it if you continue.

**Comment by beardypig** [comment]
[SCP002/xTeVe](https://github.com/SCP002/xTeVe) has been archived and hasn't received updates fo a while.

**Comment by deku-m** [comment]
> [SCP002/xTeVe](https://github.com/SCP002/xTeVe) has been archived and hasn't received updates fo a while.

Well it has more updates then the version in xteve project now. Last was just couple months back
But yes it is archived.

**Comment by michaelkrieger** [comment]
Per [this issue](https://github.com/SCP002/xTeVe/pull/44#issuecomment-1118341930) SCP002 says...

> Now, I don't plan any futurer changes into the actual source code. The only things will be updated are dockerfile and readme's (#36), but not xTeVe's internals.
> After docker version is out, planning to archivate this repository.
> 
> ...
> 
> As I've told before, it's time to switch to another project for me. And I absolutely dont mind if you fork this fork 🙃, rewire it the way you like, replace all occurences of SCP002 with Fyb3roptik and make your own build. As well as your own Docker image if you'ld like.

would be fantastic to merge these changes back into this main repo for future development as it’s come a long way but still has bugs

**Comment by Write** [comment]
A fork from SCP002's work would be gladly appreciated !

**Comment by SenexCrenshaw** [comment]
I updated SCP002's fork. There was a small bug with saving the buffer to RAM , depreciated functions and a bunch of linter fixes. 

https://github.com/SenexCrenshaw/xTeVe

**Comment by Write** [comment]
> I updated SCP002's fork. There was a small bug with saving the buffer to RAM , depreciated functions and a bunch of linter fixes.
> 
> https://github.com/SenexCrenshaw/xTeVe

Thanks a lot ! I'm trying it right now !
Can you add a way to manually set Host's IP or set a Domain ? 

Or at least set manually a custom domain when using Proxy image,
Thanks

**Comment by deku-m** [comment]
> I updated SCP002's fork. There was a small bug with saving the buffer to RAM , depreciated functions and a bunch of linter fixes.
> 
> https://github.com/SenexCrenshaw/xTeVe

There are no docker builds yet right?

**Comment by Write** [comment]
> > I updated SCP002's fork. There was a small bug with saving the buffer to RAM , depreciated functions and a bunch of linter fixes.
> > https://github.com/SenexCrenshaw/xTeVe
> 
> There are no docker builds yet right?

He updated the [Dockerfile](https://github.com/SenexCrenshaw/xTeVe/blob/master/Dockerfile), so you can easily build it yourself !

**Comment by deku-m** [comment]
Yes im not familiar with that but in the docker file it said https://hub.docker.com/r/SenexCrenshaw/xteve/
So i thought it was build.

**Comment by niallobr** [comment]
It would be good to see someone take the project forward.

The docker worked great for me, thanks for your work @SenexCrenshaw and SCP002. I had permission errors with the config folder mounted on my host (Synology). Giving 31337:31337 ownership of the folder fixed it. Maybe it can be solved by building? I just used the image.

**Comment by Write** [comment]
> It would be good to see someone take the project forward.
> 
> The docker worked great for me, thanks for your work @SenexCrenshaw and SCP002. I had permission errors with the config folder mounted on my host (Synology). Giving 31337:31337 ownership of the folder fixed it. Maybe it can be solved by building? I just used the image.

Where did you find the image ? 
Actually you'll have to build the Docker image yourself if you want @SenexCrenshaw's changes.

**Comment by SenexCrenshaw** [comment]
Next week I'll publish one on docker.

**Comment by cnorton-dev** [comment]
I'm going to attempt to start providing updates and changes based off SCP002's forked changes since it is way ahead of this repository in terms of features and abilities. I can't promise how often I can provide fixes, features and updates but will work on them as much as I can when time permits outside of my job and time with my family but as a user of xTeVe it is the least I can do to try and keep the project alive and running.

**Comment by Write** [comment]
Well I guess just simply push your changes to SenexCrenshaw's Fork ?

**Comment by SenexCrenshaw** [comment]
whatever y'all would like to do. ill try my best as well. everything seems stable for me but i am open to suggestions.

**Comment by cnorton-dev** [comment]
@SenexCrenshaw I'm open as well to anything. ~~I did hit a bug with the SCP002 2.5.0 version that I think has been reported there where some/most/all mapped channels go inactive after a scheduled update that I'm looking into~~ Turned out to be an issue with WebGrab++. I'm cool with contributing to your forked branch if you'd like or vice versa is welcome too, just hoping to keep the community and the program alive and moving along.

**Comment by SenexCrenshaw** [comment]
Just updated the README with an image based off of my fork. I fixed up the TLS implementation so you can hardcode a hostname. This fixed numerous logo issues for me.

**Comment by Write** [comment]
> Just updated the README with an image based off of my fork. I fixed up the TLS implementation so you can hardcode a hostname. This fixed numerous logo issues for me.

Thanks a lot, exactly what I needed.

**Comment by MCTyler** [comment]
No

**Comment by SenexCrenshaw** [comment]
> > It would be good to see someone take the project forward.
> > The docker worked great for me, thanks for your work @SenexCrenshaw and SCP002. I had permission errors with the config folder mounted on my host (Synology). Giving 31337:31337 ownership of the folder fixed it. Maybe it can be solved by building? I just used the image.
> 
> Where did you find the image ? Actually you'll have to build the Docker image yourself if you want @SenexCrenshaw's changes.

[https://github.com/SenexCrenshaw/xTeVe](https://github.com/SenexCrenshaw/xTeVe) has releases now and pushes multiple docker images [https://hub.docker.com/repository/docker/senexcrenshaw/xteve](https://hub.docker.com/repository/docker/senexcrenshaw/xteve)

**Comment by cnorton-dev** [comment]
> No

And why not? It's been dead for over a year now while progress and features have been added that greatly improve the program, not to mention someone has to support and fix issues and if the author is not around to do so even if it is to merge pull requests, then it needs to be changed.

**Comment by ted-gould** [comment]
So I've grabbed everyone's branches and brought mine up-to-date. Generally it seems to me that there is a group of folks that are interested in seeing xTeVe remain useful and get better. What we need now is a "root" branch that everyone agrees to send their patches to. @SenexCrenshaw are you okay with that being your branch? I'd be happy to prepare my changes as a MR on your branch.

**Comment by cnorton-dev** [comment]
Just an FYI I'm working on adding a "big" as far as code goes anyways feature that adds the ability for a user to select station logos from Tapiosinn's [tv-logos repository](https://github.com/Tapiosinn/tv-logos/). Will store the data in a JSON file in the root xteve folder with the other JSON files, it will be called tvlogos.json.

JSON file format will be as below:

`{
    "path_as_key_of_array": [
		"image_files_as_array",
		"another_image_file"
    ],
    "yet_another_path": [
		...
    ]
}`

It will get updated once a week, or we can even extend that time out further but I think once a week is a good start. This will incorporate the data list that SCP002 added to allow searching and selecting a logo from the list.

Code wise it's all done except for adding the picker and data list to the channel mapping popup for the Logo URL textbox.

Going to see what I can do about adding a preview for the logo as well but haven't gotten to that point yet.

And for those wondering yes this is because of a frustration I personally have with having to get the logo in the repository, click on download, copy the URL, go back to xteve, paste and save, rinse and repeat for lots of channels.... yes I could download them but then I would still have to click upload logo, find the logo in the folder, "upload" it, save, rinse and repeat.

Why do all that when I can just start typing a TV station name in a textbox and it popup with logos that match and click on one I need and be done.

**Comment by SenexCrenshaw** [comment]
Whatever y'all want to do. I have been happy with the fixes so far. Let me know how i can help.

**Comment by SenexCrenshaw** [comment]
> Just an FYI I'm working on adding a "big" as far as code goes anyways feature that adds the ability for a user to select station logos from Tapiosinn's [tv-logos repository](https://github.com/Tapiosinn/tv-logos/). Will store the data in a JSON file in the root xteve folder with the other JSON files, it will be called tvlogos.json.
> 
> JSON file format will be as below:
> 
> `{ "path_as_key_of_array": [ "image_files_as_array", "another_image_file" ], "yet_another_path": [ ... ] }`
> 
> It will get updated once a week, or we can even extend that time out further but I think once a week is a good start. This will incorporate the data list that SCP002 added to allow searching and selecting a logo from the list.
> 
> Code wise it's all done except for adding the picker and data list to the channel mapping popup for the Logo URL textbox.
> 
> Going to see what I can do about adding a preview for the logo as well but haven't gotten to that point yet.
> 
> And for those wondering yes this is because of a frustration I personally have with having to get the logo in the repository, click on download, copy the URL, go back to xteve, paste and save, rinse and repeat for lots of channels.... yes I could download them but then I would still have to click upload logo, find the logo in the folder, "upload" it, save, rinse and repeat.
> 
> Why do all that when I can just start typing a TV station name in a textbox and it popup with logos that match and click on one I need and be done.

Didnt realize Tapiosinn/tv-logos existed! I would like to help with your PR if you would like. 

Also I think we shouldnt muck up this thread anymore. We can take these convos to another fork/repo. mine is fine if you want

**Comment by MCTyler** [comment]
xTeVe is "up to date" and working fine. Use all these other forks at your own risk.

**Comment by MCTyler** [comment]
> > No
> 
> And why not? It's been dead for over a year now while progress and features have been added that greatly improve the program, not to mention someone has to support and fix issues and if the author is not around to do so even if it is to merge pull requests, then it needs to be changed.

Not Dead, just no revision needed.

**Comment by MCTyler** [comment]
Join us on our official Discord Channel if you have any further questions.....
https://discord.gg/KD2jy5V

**Comment by cnorton-dev** [comment]
> Not Dead, just no revision needed.

To be honest there is a lot that could be improved upon and that a revision should be considered.

Such as buffering, which with SCP002's fork adds memory buffering which outperforms the buffering to the drive, reading that buffer and sending it to the client which I can tell on the client when the buffer gets switched to the next buffer file as there is a slight pause as the new buffer is fed to the client.

TLS mode is another feature that is a great addition

The feature I added which allows you to enable "shift channels" which allows you to move a channel to an already occupied channel without it going all the way to the next open channel but instead replaces that channel and shifts the existing channel and all others after up a number until the end of the list or an empty channel slot is found.

Selecting the host IP that will be written into the files is amazing and I don't have to remember to use my network IP address and can use localhost and it not completely screw over the XML and m3u files.

General code cleanup.

I'm sure you catch my drift by now. If the decision is none of that is worth another revision, that's fine, but to say use other forks at your own risk when we as a community are improving upon something that appears to no longer be an active project is a way to have people not want to contribute to the main project anymore since, well, we can't anyways because pull requests are no longer getting reviewed and merged.

Not to mention the issues list on here isn't even getting addressed anymore.

**Comment by MCTyler** [comment]
I don't see any features you suggest that would interest me or other staff. Using code that opens your connection to WWW always has inherent risks. That's just a generic warning.

**Comment by MCTyler** [comment]
As far as buffering, Most Media Players already buffer streams and i find this unnecessary.

**Comment by Fyb3roptik** [comment]
I have been redoing xteve for the last month or so. I will post the link here when its stable. I have added some things like multi-select bulk edit, setting starting channel number for a filter group or multi-selected channels, new UI, each page has its own link now (no more one page app), new updater, docker support.

**Comment by deku-m** [comment]
I really hope you are combining all your fixes and new features in one branch that will bring this app back to life.

**Comment by Fyb3roptik** [comment]
It will be in a diff repo and rebranded. I will post it up as soon as I am able to fix up my BETA

**Comment by Fyb3roptik** [comment]
https://github.com/Threadfin/Threadfin/releases/tag/BETA-0.8.1

**Comment by Fyb3roptik** [comment]
While it does say BETA it merely adds some functionality and features to the existing xteve, so it should be stable

**Comment by Fyb3roptik** [comment]
@SenexCrenshaw Care to help me get the RAM buffer stuff added? What was it that needed fixing? I tried adding it earlier but it wasnt working

**Comment by SenexCrenshaw** [comment]
Sure thing. It was switched to using a vfs filesystem/buffer.
Are you still using go Lang? I wanted to port to c# core and vue. Js but the patches in my repo work without a huge uplift so I haven't cared to spend the time.


If you want, send me your beta branch link and I'll add it in maybe.

If anything though. I think we should move these discussions to a different repo. 


The work on the original xteve repo is excellent and this thread should be closed.

**Comment by Fyb3roptik** [comment]
Go will be much faster than C# so yeah still in Go. I added his changes in a previous commit but yeah it wouldnt work no matter what I tried, client would always disconnect.

**Comment by cnorton-dev** [comment]
@Fyb3roptik See my fix in @SenexCrenshaw fork that fixed VFS for Windows due to a panic error in relation to the drive volume not existing if any drive other than C:\ is used for the cache location set in xTeVe.

**Comment by Fyb3roptik** [comment]
https://github.com/Threadfin/Threadfin/tree/beta

**Comment by Fyb3roptik** [comment]
> @Fyb3roptik See my fix in @SenexCrenshaw fork that fixed VFS for Windows due to a panic error in relation to the drive volume not existing if any drive other than C:\ is used for the cache location set in xTeVe.

I am not on Windows, using Ubuntu but will look at it for sure. PR's are always welcome!

**Comment by cnorton-dev** [comment]
Cool, yeah there isn't any issues with the VFS and *nix OS, just some weirdness with Windows that has to be covered, ya know typical Windows things but adding it in should be pretty straight forward. I'm working on another feature currently once I get some actual time outside of work programming but I'm sure either me or SenexCrenshaw can help you out with getting it added in.

**Comment by SenexCrenshaw** [comment]
Sure thing. It was switched to using a vfs filesystem/buffer.
Are you still using go Lang? I wanted to port to c# core and vue. Js but the patches in my repo work without a huge uplift so I haven't cared to spend the time.


If you want, send me your beta branch link and I'll add it in maybe.

If anything though. I think we should move these discussions to a different repo. 


The work on the original xteve repo is excellent and this thread should be closed.

**Comment by Fyb3roptik** [comment]
Here are my changes:
* Brand new Bootstrap based UI
* Separate table for inactive channels
* Multi-select checkboxes by holding shift
* Starting channel by Filter Group
* Starting Channel by Bulk Edit channels
* PPV Channel name mapping to EPG
* Docker support
* Channels now sort properly by name

**Comment by cnorton-dev** [comment]
I've added discussions to my fork so we can move conversations there that way this ticket can be closed or what not: https://github.com/cnorton-webdev/xTeVe/discussions

**Comment by Fyb3roptik** [comment]
https://github.com/Threadfin/Threadfin/discussions

Lets do it in mine :)

**Comment by SenexCrenshaw** [comment]
Faster? This is a m3u proxy. There is nothing in here besides the kludgy get next channel number function that has to be preformant. Even that logic is off.

If this is about how to shove 448 channels into plex then who cares about editing speed. It's about features and maintenance.

**Comment by Fyb3roptik** [comment]
Go > C# but I am biased lol

**Comment by Fyb3roptik** [comment]
https://github.com/Threadfin/Threadfin/discussions/7

**Comment by GwynethLlewelyn** [comment]
Sorry, guys, but I'm thoroughly confused now :)

It's not unusual for communities to split their forks following some disagreement on the way the software should take in the future — but I wonder if there couldn't be some sort of agreement on _where_ development shall continue? I see that there are already a few people who are busy merging changes from three different forks (in order to get benefits from all of them!). It's getting a bit hard to follow :)

Anyway, it was just a rant. I like xTeVe a lot, but I'm also well aware that some polishing is necessary, some bugs require fixing, and there is a plethora of goodies out there, that didn't exist when the project started, which could very well be added to the 'core' project. Nothing against such changes (especially when they are for the better!).

(On the Go vs. C# issue, like @Fyb3roptik, I'm seriously biased — I do some tinkering in Go and am _very_ fond of it, since it's a plain and simple language but with an overwhelming number of incredible features for those who need them; while C#, well, is a rotten clone of C++ — which, in itself, grew to become a monster, completely out of control — with some winks at Java developers, and, in spite of Microsoft's magnificent and blazingly fast Roslyn compiler, remains a programming language for masochists. Go is what I use by _choice_ — probably the first language I've chosen for myself since the 1986 version of C++, all others having been mostly pushed upon me to maintain legacy projects)

I know — I'm certainly digressing; my point is that I'll be eagerly following those forks that keep xTeVe in good, old, easily-to-compile-on-any-platform Go — which I do on low-end devices such as the Raspberry Pi (aye, I know that cross-compilation _exists_, but what's the fun of that, as opposed to seeing underpowered devices with little memory actually finishing a compilation in a decent amount of time and getting _working_ binaries? 😁 ). I can't do it as easily with C#. It's true that someone might port xTeVe to, say, Node.js (a popular choice these days), or be happy to write three times the number of lines just to get it compiled on Rust (and gloat about the 0.1% performance they get over Go), but, alas, why bother? xTeVe works well under Go, so let it remain a Go project, and feel free to start a _new_ project using whatever programming language you believe that has a chance of doing things better, faster, and easier to maintain (good luck on all three)...

**Comment by Fyb3roptik** [comment]
Check out my Threadfin repo. I redid xteve. The beta branch has the changes and I have docker

**Comment by MCTyler** [comment]
As far as we're concerned, development is complete. If you want something different, go to a fork. @GwynethLlewelyn

**Comment by GwynethLlewelyn** [comment]
@MCTyler fair is fair! Or do my own fork 😀 I cannot disagree on that approach — but... there are always bugs, fixes, security concerns, improvements on things that used to work well but now don't, etc. and so forth.
@Fyb3roptik I did, [I did](https://github.com/Threadfin/Threadfin/discussions/6#discussioncomment-4485641) :-) I personally I'm not so fond of running Docker on slow, underperforming systems; I rather prefer running it directly.
Sadly, at least for me, your _current_ version (not the beta!) seems to suffer from similar problems to those from the @MCTyler original version (e.g. crash when reading the XEPG database) — I guess I'll have to be patient and wait until the beta is fully merged into the master branch.
Ironically, @SenexCrenshaw has a rather fully functional version of xTeVe. I say 'ironically' because AFAIK @SenexCrenshaw's project will abandon Go and move ahead with a port from scratch in C#...

**Comment by MCTyler** [comment]
Your talking about the lack of ability of a browser, not xTeVe crashing. That can be solved by making the user interface not Browser dependent. As far as Beta Verses any other version, they are both the same. Sure, Program in the language you feel comfortable with!
Here's a fork we had good reviews on.....https://github.com/SCP002/xTeVe

**Comment by Fyb3roptik** [comment]
Use the beta not main. It's about to go 1.0 and had tons of changes and I don't use docker either, I just have it as an option

**Comment by Fyb3roptik** [comment]
Also who in their right mind chooses a slower dotnet over Go?! 🤣

**Comment by MCTyler** [comment]
> Use the beta not main. It's about to go 1.0 and had tons of changes and I don't use docker either, I just have it as an option

Would you please not use xTeVe as a forum for your fork. It gets confusing. I'm answering people about xTeVe not realizing they are talking about your fork. Thanks!

**Comment by SenexCrenshaw** [comment]
> @MCTyler fair is fair! Or do my own fork 😀 I cannot disagree on that approach — but... there are always bugs, fixes, security concerns, improvements on things that used to work well but now don't, etc. and so forth. @Fyb3roptik I did, [I did](https://github.com/Threadfin/Threadfin/discussions/6#discussioncomment-4485641) :-) I personally I'm not so fond of running Docker on slow, underperforming systems; I rather prefer running it directly. Sadly, at least for me, your _current_ version (not the beta!) seems to suffer from similar problems to those from the @MCTyler original version (e.g. crash when reading the XEPG database) — I guess I'll have to be patient and wait until the beta is fully merged into the master branch. Ironically, @SenexCrenshaw has a rather fully functional version of xTeVe. I say 'ironically' because AFAIK @SenexCrenshaw's project will abandon Go and move ahead with a port from scratch in C#...

https://github.com/SenexCrenshaw/StreamMaster

**Comment by Fyb3roptik** [comment]
> > Use the beta not main. It's about to go 1.0 and had tons of changes and I don't use docker either, I just have it as an option
> 
> Would you please not use xTeVe as a forum for your fork. It gets confusing. I'm answering people about xTeVe not realizing they are talking about your fork. Thanks!

Xteve is dead.
